tests/no_cycles.rb

```diff
@@ -0,0 +1,50 @@
+require 'find'
+
+@all_pkgs = {}
+
+# Loads all packages
+Find.find("#{CUR_DIR}/../packages") do |filename|
+  if File.extname(filename) == '.rb'
+    name = File.basename(filename, '.rb')
+    require_relative("../packages/#{name}")
+    pkg = Object.const_get(name.capitalize)
+    pkg.name = name
+    @all_pkgs[name] = pkg
+  end
+end
+
+# Looking for cycles. @path will keep the current dependency path.
+# @state will store :on_path for vertices on the current dependency path
+# and :visited for vertices that have already been checked not to lead to
+# cycles.
+@state = {}
+@path = []
+def dfs(pkg)
+  @path.push(pkg.name)
+  if @state[pkg] == :on_path
+    puts "Found dependency cycle!".lightred
```

Comment:
Should this be abort instead of puts?  Otherwise, it continues on and you get the success message at the end, correct?

Reply:
Never mind.  I see you have the exit 1 at the end.

Reply:
Yes, I still need to print the dependency cycle.